core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncRetryableStage.java

```diff
@@ -128,6 +132,7 @@ private void retryResponseIfNeeded(Response<OutputT> resp, CompletableFuture<Res
                 }
 
                 if (shouldRetry(resp.httpResponse(), resp.exception())) {
+                    responseHandler.onError(err);
```

Comment:
Can we add a comment about the last error being delivered in the client? That's a bit tricky and not obvious from the code.